src/client/common/cancellation.ts

```diff
@@ -14,6 +14,12 @@ export class CancellationError extends Error {
         super(localize.Common.canceled());
     }
 }
+
+/**
+ * Error type thrown when a timeout occurs
+ */
+export class TimedOutError extends Error {}
```

Comment:
This is a bit of an odd spot for this class, it's not cancellation related. If it's specific to the kernel launcher I'd put it as KernelTimedOutError and put it in the class that throws it. Or if it's general, maybe errorUtils or a types.ts file?